hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/MergeIntoHoodieTableCommand.scala

```diff
@@ -213,6 +221,26 @@ case class MergeIntoHoodieTableCommand(mergeInto: MergeIntoTable) extends Hoodie
     }
   }
 
+  /**
+   * Check whether expression of preCombine field have the same column name with target column.
+   *
+   * Merge expression cases that return true:
+   * 1) merge into .. on .. update set ts = s0.ts
+   * 2) merge into .. on .. update set ts = cast(s0.ts as int)
+   * 3) merge into .. on .. update set ts = s0.ts+1 (expressions like this whose sub node has the same column name with target)
+   * "ts" is preCombine field of h0.
+   */
+  private def isEqualToTargetPreCombineField(targetColumnName: String, sourceExpression: Expression): Boolean = {
```

Comment:
I think we can improve the naming of this method, currently it's not self-contained -- i have to go and see how it's used to understand what it does.

Let's rename 
  - This one to `containsPreCombineFieldReference`
  - `isEqualTo` one to `containsPrimaryKeyFieldReference`

Reply:
I tried to improve the name of these two methods. How about this:
`isEqualTo` to `isSameColumnName`
`isEqualToTargetPreCombineField` to `containsSameColumnNameForPreCombine`

Because `isEqualTo` is for columns in merge condition, maybe not only primaryKey.
And both of the two methods are to find whether column name is equal, not whether field is contained.

Reply:
Let's take a step back: what's the goal of this method? It's not just comparing whether the column is the same but rather sifting t/h nested expressions looking for `AttributeReference` that refers to the target column -- hence my suggestion to call it contains b/c it traverses the expression tree. When we call it `isSomethingEqual` it leaves an impression that we're just check the equality which is misleading in that case

Reply:
Thanks for explaining, I agree with you.
I have changed it with your suggestion above.